lib/logflare_web/controllers/api/access_token_controller.ex

```diff
@@ -0,0 +1,73 @@
+defmodule LogflareWeb.Api.AccessTokenController do
+  use LogflareWeb, :controller
+  use OpenApiSpex.ControllerSpecs
+
+  alias Logflare.Auth
+  alias LogflareWeb.OpenApi.Accepted
+  alias LogflareWeb.OpenApi.Created
+  alias LogflareWeb.OpenApi.List
+  alias LogflareWeb.OpenApi.NotFound
+  alias LogflareWeb.OpenApi.Unauthorized
+  alias LogflareWeb.OpenApiSchemas.AccessToken
+  alias LogflareWeb.Api.FallbackController
+
+  action_fallback(FallbackController)
+
+  tags(["management"])
+
+  operation(:index,
+    summary: "List access tokens",
+    responses: %{200 => List.response(AccessToken)}
+  )
+
+  def index(%{assigns: %{user: user}} = conn, _) do
+    tokens = Auth.list_valid_access_tokens(user) |> Enum.map(&maybe_redact_token/1)
+    json(conn, tokens)
+  end
+
+  operation(:create,
+    summary: "Create source",
+    request_body: AccessToken.params(),
+    responses: %{
+      201 => Created.response(AccessToken),
+      401 => Unauthorized.response(),
+      404 => NotFound.response()
+    }
+  )
+
+  def create(%{assigns: %{user: user}} = conn, params) do
+    scopes_input = Map.get(params, "scopes", "")
+
+    with {:scopes, true} <- {:scopes, not (scopes_input =~ "partner")},
+         {:ok, access_token} <- Auth.create_access_token(user, params) do
+      conn
+      |> put_status(201)
+      |> json(access_token)
+    else
+      {:scopes, false} -> {:error, :unauthorized}
+    end
```

Comment:
```suggestion
    if "partner" in String.split(scopes_input) do
      {:error, :unauthorized}
    else
      {:ok, access_token} = Auth.create_access_token(user, params)
      
      conn
      |> put_status(201)
      |> json(access_token)
    end
```

Reply:
this would result in bad params not propagating up.

Reply:
It was not propagating previously either, as there was no `else` clause that would match incorrect return from `Auth.create_access_token/2`.

Reply:
adjusted to forward changeset errors